### ISSUE TITLE
Build fails if .so files are put in the "lib" folder 

### ISSUE THREAD

**Issue by david-bouyssie** [scala-native/scala-native#1711]
Compilation fails when .so files are copied in the "lib" folder (SBT's unmanaged dependencies folder, see https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html).

The error is similar to the one described in this previous issue: https://github.com/scala-native/scala-native/issues/911

```java.lang.UnsupportedOperationException: Neither a jar, nor a directory```

The solution can consist in using a different directory or to change the SBT's "unmanaged dependencies folder". For instance this additional settings will allow to use the "lib" folder to contain .so files:
```unmanagedBase := baseDirectory.value / "jars"```

This behavior should be described in the SN FAQ or in a new troubleshooting section.

**Comment by errikos** [comment]
An alternative would be to use `nativelib` instead of `lib` for Scala Native.

**Comment by ekrich** [comment]
I would like to go ahead and fix this problem but I would like to get some input.

As mentioned in this issue #911 is very similar to this issue. Both occur deep in the build process here as follows:
https://github.com/scala-native/scala-native/blob/master/util/src/main/scala/scala/scalanative/io/VirtualDirectory.scala#L65

Or in the case of the first problem where it was a directory but empty.
https://github.com/scala-native/scala-native/blob/master/util/src/main/scala/scala/scalanative/io/VirtualDirectory.scala#L171

In the case of the first PR we filtered the class path.
https://github.com/scala-native/scala-native/blob/master/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala#L122

I would prefer to not add more logic to the plugin as that makes integrators repeat the logic. The classpath goes directly into the config object which has no logic. We also have Discovery do some work so people can reuse that code. A couple of points.

1. We might want to support putting C/C++ libraries or `.o` files in the `lib` directory and adding them to the link phase in the future.
2. It would be nice if the logic is internal so we could change in the future if desired and not expose the details.

It would be great if we just changed the classpath in the config object before the config object gets passed around before here but that feels kind of wrong. Also adding logic to the config object feels wrong.
https://github.com/scala-native/scala-native/blob/master/tools/src/main/scala/scala/scalanative/build/Build.scala#L56

The other fix is down if the VirtualDirectory but that seems a bit weird as that code doesn't really need to know about weird classpath items. 

I feel like we should filter for both items in `Build.scala` changing the config but wondering what people think.
